jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp

```diff
@@ -38,19 +41,259 @@
  */
 
 /**
+ * Helper function to try to parse a part of a date string
+ *
+ * @return NaN if cannot read from string, ToNumber() otherwise
+ */
+static ecma_number_t
+ecma_date_parse_date_chars (lit_utf8_iterator_t *iter, /**< iterator of the utf8 string */
+                            uint32_t num_of_chars) /**< number of characters to read and convert */
+{
+  ecma_number_t ret_value = ecma_number_make_nan ();
```

Comment:
Also, I would ask you to rename the variable from `ret_value` to something like `ret_num`. `ret_value` is, currently, mostly used for completion values, so can be confusing here.